api/app/Models/User.php

```diff
@@ -95,8 +97,52 @@ class User extends Model implements Authenticatable, LaratrustUser
     protected $fillable = [
         'email',
         'sub',
+        'searchable',
     ];
 
+    protected $hidden = [
+        'searchable',
+    ];
+
+    /**
+     * Get the indexable data array for the model.
+     *
+     * @return array<string, mixed>
+     */
+    public function toSearchableArray(): array
+    {
+        $this->loadMissing([
+            'poolCandidates',
+            'workExperiences',
+            'educationExperiences',
+            'personalExperiences',
+            'communityExperiences',
+            'awardExperiences',
+        ]);
+        $attributesToPluck = [
+            'poolCandidates' => ['notes'],
+            'workExperiences' => ['role', 'organization', 'division', 'details'],
+            'educationExperiences' => ['thesis_title', 'institution', 'details', 'area_of_study'],
+            'personalExperiences' => ['title', 'description', 'details'],
+            'communityExperiences' => ['title', 'organization', 'project', 'details'],
+            'awardExperiences' => ['title', 'details', 'issued_by'],
+        ];
+        $entireIndex = collect([$this->email, $this->first_name, $this->last_name, $this->telephone, $this->current_province, $this->current_city]);
+        foreach ($attributesToPluck as $relationship => $attributes) {
+            foreach ($attributes as $attribute) {
+                $values = $this->$relationship->pluck($attribute)->reject(function ($value) {
+                    return is_null($value) || $value === '';
+                })->toArray();
+
+                // Merge non-null and non-empty values
+                $entireIndex = $entireIndex->merge($values);
+            }
+        }
```

Comment:
I reverted the changes         
```
return collect([$this->email, $this->first_name, $this->last_name, $this->telephone, $this->current_province, $this->current_city])
            ->merge($this->poolCandidates()->pluck('notes'))
            ->merge($this->workExperiences()->pluck('role'))
            ->merge($this->workExperiences()->pluck('organization'))
            ->merge($this->workExperiences()->pluck('division'))
            ->merge($this->workExperiences()->pluck('details'))
            ->merge($this->educationExperiences()->pluck('thesis_title'))
            ->merge($this->educationExperiences()->pluck('institution'))
            ->merge($this->educationExperiences()->pluck('details'))
            ->merge($this->educationExperiences()->pluck('area_of_study'))
            ->merge($this->personalExperiences()->pluck('title'))
            ->merge($this->personalExperiences()->pluck('description'))
            ->merge($this->personalExperiences()->pluck('details'))
            ->merge($this->communityExperiences()->pluck('title'))
            ->merge($this->communityExperiences()->pluck('organization'))
            ->merge($this->communityExperiences()->pluck('project'))
            ->merge($this->communityExperiences()->pluck('details'))
            ->merge($this->awardExperiences()->pluck('title'))
            ->merge($this->awardExperiences()->pluck('details'))
            ->merge($this->awardExperiences()->pluck('issued_by'))
            ->reject(function ($value) {
                return is_null($value) || $value === '';
            })->toArray();

```
because 
1. It broke the bunch of tests in AuthProvider as there are some nulls 
 Reject after all collection merge doesn't affect the nulls in the individual collection. It affects only the merged array.
Ex. If there are null or empty values in the notes, they won't be removed by the reject function in this context. 
Thats why it broke the test. 
2. Actually It is better practice to remove nulls before merging in, instead of removing it after the fact.

Reply:
Refactored in https://github.com/GCTC-NTGC/gc-digital-talent/pull/8539/commits/770be873d1d04173b29709365c728a3523444711